### ISSUE TITLE
Which window responds to keyboard messages?

### ISSUE THREAD

**Issue by fran-f** [microsoft/terminal#6359]
If I were sending `WM_KEYDOWN` / `WM_KEYUP` messages to Windows Terminal (for example to simulate `Ctrl-Shift-1`) which window class should I target? I have tried with no luck with `CASCADIA_HOSTING_WINDOW_CLASS`, and I'm a bit lost around the XAML bits.

**Comment by DHowett** [comment]
Depends on what you're trying to accomplish! What are you trying to accomplish?

**Comment by fran-f** [comment]
I'm writing a tool to open new Terminal tabs, since at the moment it's not possible from the command line. The idea was to take the top Terminal window, and send the correct key combination for the profile I want to open (I can get both the profile and keyboard shortcut from the settings file)

**Comment by zadjii-msft** [comment]
Wait, are you trying to create a commandline application to be able to run in the Terminal to send keystrokes corresponding to Terminal keybindings to the Terminal _isntead of just pressing that key_?

**Comment by fran-f** [comment]
I'm not, but that would be a good idea! Because then I could create shortcuts to that application to open any profile as a tab. I'm actually working on a [plugin](https://github.com/fran-f/keypirinha-terminal-profiles) for a [keyboard launcher](keypirinha.com/).

I can open a profile in a new window, but I'd like to check if a Terminal is already open, and in that case open a new tab in the same window.

**Comment by fran-f** [comment]
After a bit of digging I found `CASCADIA_HOSTING_WINDOW_CLASS` ⭢ `Windows.UI.Input.InputSide.WindowClass`. Sending keystrokes here is equivalent to typing in the shell.

However, Terminal keybinding don't have effect. Sending `Ctrl-V` shows `^V` at the prompt, even with this configuration in place:
```
    "keybindings": [
            { "command": "paste", "keys": "ctrl+v" }
    ]
```

So that's not the right place either ¯\_(ツ)_/¯

**Comment by DHowett** [comment]
So, it _should_ be either the `CASCADIA_WINDOW_HOSTING_CLASS` (in that it's the toplevel window that receives all input) or the input site. The input site is an implementation detail that we don't control, whereas the cascadia class name is one we _do_ control.

Are you sending `WM_KEYDOWN` or `WM_CHAR`? Or both?

I can't _support_ you doing this insofar as I won't promise to not break you, but I will _help_ :smile:

(Closing, but let's keep the discussion going.)

**Comment by fran-f** [comment]
Thanks for the unofficial support 🙂 
I tried with a few different methods, always sending `WM_KEYDOWN` and `—UP` combinations. Manually, and using [pywinauto](https://pywinauto.readthedocs.io/), a win32 automation library.

The closes I got is sending events to `CASCADIA_WINDOW_HOSTING_CLASS` as you suggested. Sending the sequence `Shift-1: '+1'  Ctrl-V: '^v'  Ctrl-Shift-V: '^+v'  Ctrl-Shift-2: '+2'` gives me this:
![image](https://user-images.githubusercontent.com/50931058/86518276-527b4100-be28-11ea-8d21-1361ff532ee0.png)

Close, but `Ctrl-Shift-x` sequences seem to fail. However, if I also bind `Ctrl-V` to `paste`, there is no output and no effect.

This route is probably more frail and complex than I thought initially, so I'm tempted to abandon it. I'll wait the official way to create new tabs in an existing window.
